code/modules/projectiles/guns/projectile/shotgun/doublebarrel.dm

```diff
@@ -84,16 +84,20 @@
 
 /obj/item/weapon/gun/projectile/shotgun/doublebarrel/attackby(var/obj/item/A as obj, mob/user as mob)
 	if(QUALITY_SAWING in A.tool_qualities)
-		to_chat(user, SPAN_NOTICE("You begin to shorten the barrel of \the [src]."))
-		if(loaded.len)
-			for(var/i in 1 to max_shells)
-				afterattack(user, user)	//will this work? //it will. we call it twice, for twice the FUN
-				playsound(user, fire_sound, 50, 1)
-			user.visible_message(SPAN_DANGER("The shotgun goes off!"), SPAN_DANGER("The shotgun goes off in your face!"))
-			return
-		if(A.use_tool(user, src, WORKTIME_FAST, QUALITY_SAWING, FAILCHANCE_NORMAL, required_stat = STAT_COG))
-			qdel(src)
-			new /obj/item/weapon/gun/projectile/shotgun/doublebarrel/sawn(usr.loc)
-			to_chat(user, SPAN_WARNING("You shorten the barrel of \the [src]!"))
+		if (src.name != "sawn-off shotgun")
```

Comment:
You might have use `!istype(src, /obj/item/weapon/gun/projectile/shotgun/doublebarrel/sawn)` since it does check if the weapon is a sawn or not.

Reply:
Got it, let me look into options.

Reply:
Got it, let me look into options of gun mods, I thought they are usually destroyed on removal. I will revise this.

Reply:
1) Fixed so the check is based on type, not name.
2) Added a check for attachments, and if any are present, the user gets a query if he wants to destroy them or not ("No" cancels the action, "Yes" proceeds).